pallets/pools/src/lib.rs

```diff
@@ -1491,22 +1234,23 @@ pub mod pallet {
 			};
 			let order =
 				Order::<T>::try_get(&loc, &who).map_err(|_| Error::<T>::NoOutstandingOrder)?;
+
+			let end_epoch: T::EpochId = collect_n_epochs
+				.checked_sub(&One::one())
+				.ok_or(Error::<T>::CollectsNoEpochs)?
+				.checked_add(&order.epoch)
+				.ok_or(DispatchError::from(ArithmeticError::Overflow))?;
+
 			ensure!(
-				order.epoch <= pool.last_epoch_executed,
+				end_epoch <= pool.last_epoch_executed,
```

Comment:
Why was this changed to `end_epoch`? I think this should remain `order.epoch`.

Reply:
We are checking here if the last epoch a user wants to collect is already executed. Or am I missing something here?

Reply:
If they submitted an order in epoch 2, the current `last_epoch_executed` is 2, and they want to collect 1 epoch, then `end_epoch` is 3. Now, `end_epoch <= pool.last_epoch_executed` is false, but this should be possible, while `order.epoch <= pool.last_epoch_executed` is correctly true.

Reply:
Ahh, but https://github.com/centrifuge/tinlake/blob/develop/src/lender/tranche.sol#L210 meant that the Solidity implementation had the same effect, but was more implicit. I think you're right that by changing this to `end_epoch`, the code has the right behaviour.

Reply:
Took another look at this.

Say an investor submitted orders in epoch 2 and 3, and epoch 2 was executed, epoch 3 wasn't.
In this case, collect 1 epoch should work, collect 2 epochs should fail.

```
order.epoch = 2
last_epoch_executed = 2

collect_n_epochs = 1
order.epoch <= last_epoch_executed = 2 <= 2 = true
end_epoch = 1 - 1 + 2 = 2
end_epoch <= last_epoch_executed = 2 <= 2 = true

collect_n_epochs = 2
order.epoch <= last_epoch_executed = 2 <= 2 = true
end_epoch = 2 - 1 + 2 = 3
end_epoch <= last_epoch_executed = 3 <= 2 = false
```

So the new implementation using `end_epoch <= last_epoch_executed` is indeed correct. The Solidity implementation worked because `end_epoch` would be set to `last_epoch_executed` on https://github.com/centrifuge/tinlake/blob/develop/src/lender/tranche.sol#L212.